json-body-transformer/src/test/java/com/ninecookies/wiremock/extensions/JsonBodyTransformerTest.java

```diff
@@ -309,11 +333,21 @@ public void doesNotApplyGlobally() throws Exception {
 		wireMockServer.verify(postRequestedFor(urlEqualTo(REQUEST_URL)));
 	}
 
-	@Test
-	public void injectRandom() {
+	@DataProvider
+	private Object[][] randomFormats() {
+		return new Object[][] {
+				{ "$(!Random)" },
+				{ "$(!Random.1)" },
+				{ "$(!Random.abc)" },
+				{ "$(!Random.ABC)" },
+		};
+	}
+
+	@Test(dataProvider = "randomFormats")
+	public void injectRandom(String format) {
 		wireMockServer.stubFor(post(urlEqualTo(REQUEST_URL))
 				.willReturn(aResponse().withStatus(200).withHeader("content-type", CONTENT_TYPE)
-						.withBody("{\"randomNumber\":\"$(!Random)\", \"got\":\"it\"}")
+						.withBody("{\"randomNumber\":\"" + format + "\", \"got\":\"it\"}")
```

Comment:
Nop. The intention behind the suffixes is to distinguish between different properties.
Imagin following JSON
```JSON
{
  "id": "$(!UUID)",
  "loginId": "$(!UUID)"
}
```
The `$(!UUID)` is the key in the mapping where the generated value is assigned to. Afterwards all occurrences of `$(!UUID)` are replaced with the according value.
The suffix enables one to get two different UUIDs generated by specifying
```JSON
{
  "id": "$(!UUID.1)",
  "loginId": "$(!UUID.2)"
}
```
instead.
It will be more clear if you take a look at `injectMultipleRandomsWithReuse()` and the readme... ;)